python/foglamp/plugins/south/cc2650poll/sensortag_cc2650.py

```diff
@@ -140,35 +140,32 @@ class SensorTagCC2650(object):
     _CHAR_HANDLE_TIMEOUT = 3
 
     def __init__(self, bluetooth_adr, timeout):
-        try:
-            self.bluetooth_adr = bluetooth_adr
-
-            # If "con" var is set at class level, pick that one else create a new instance
-            if SensorTagCC2650.con is None:
-                self.con = pexpect.spawn('gatttool -b ' + bluetooth_adr + ' --interactive')
-            else:
-                self.con = SensorTagCC2650.con
-
-            self.con.expect('\[LE\]>', timeout=int(timeout))
-
-            msg_debug = 'SensorTagCC2650 {} Connecting... If nothing happens, please press the power button.'.\
-                        format(self.bluetooth_adr)
-            print(msg_debug)
-            _LOGGER.debug(msg_debug)
-
-            self.con.sendline('connect')
-            self.con.expect('.*Connection successful.*\[LE\]>', timeout=int(timeout))
-            self.is_connected = True
-            msg_success = 'SensorTagCC2650 {} connected successfully'.format(self.bluetooth_adr)
-            print(msg_success)
-            _LOGGER.debug(msg_success)
-        except (ExceptionPexpect, EOF, TIMEOUT, Exception) as ex:
-            self.is_connected = False
-            self.con.sendline('quit')
-            # TODO: Investigate why SensorTag goes to sleep often and find a suitable software solution to awake it.
-            msg_failure = 'SensorTagCC2650 {} connection failure. Timeout={} seconds.'.format(self.bluetooth_adr, timeout)
-            print(msg_failure)
-            _LOGGER.exception(msg_failure)
+        attempt_count = 1
+        while True:
```

Comment:
this will never die? can we make this attempt configurable? may be 100 times as default

Reply:
we should have some attempt count, may be 100 or 1000 but not infinite